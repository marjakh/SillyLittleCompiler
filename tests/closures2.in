function plusser_generator(x) {
  function inner(y) {
    return y + x;
  }
  return inner;
}

let plus5 = plusser_generator(5);
print(plus5(4));
print(plus5(8));

let plus3 = plusser_generator(3);
print(plus3(4));
print(plus3(8));