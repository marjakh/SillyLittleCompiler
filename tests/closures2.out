9
13
7
11
